ci: fix #29

nouex opened the pull request:
This is a work-in-progress PR for #29.  It is currently under investigation and as the bug is detected on [CI](https://travis-ci.org/reecehudson/node-emitter-details) this might have some invoke pushes to for the sole purpose of checking the CI log.

reecehudson commented:
After the last commit, our wanted check did not appear to run, [rerun](https://travis-ci.org/reecehudson/node-emitter-details/builds/139346345).

reecehudson commented:
Fuck!!!  Still not running our wanted check, i.e. `node ./test-files.js`

Commit:
.travis.yml: remove unused env variable and add new one

Commit:
test, .travis.yml: count tests on CI

Since we are manually adding tests on .travis.yml to the `script` field it helps
to automate checking that you did not forget to add a new test to the list.

Commit:
ci: pass ci

Commit:
ci: enable debug(travis-count-tests) logging

Commit:
ci: don't build per env var

Commit:
ci: fix err, invoke ci

Commit:
ci: fix tiny err, invoke ci

Commit:
fix, invoke ci

Commit:
invoke ci

Commit:
ci: make ci pass

Commit:
test, travis-count-tests.js, tot-tested.js: switch from using env var to require()

Commit:
test: log count

Commit:
Revert "test: log count"

This reverts commit eb82220b1521406ebe2f43cec9e32317d8d51602.

Commit:
Revert "test, travis-count-tests.js, tot-tested.js: switch from using env var to require()"

This reverts commit 287bcfd07920988c1eaa7fde070467f07a7c6b3b.

Commit:
.travis.yml: try global env vars

reecehudson commented:
So as you can see the past 10 or so commits have been devoted to fixing tweaks here and there and invoking CI.  For now the only (and last) thing stopping me from a green CI is the ability to have an env variable maintain state as it is changed throughout all builds (more specifically to our case, that is the counter of files that have been tested).

I don't see a way through this barrier.  Closing this and going to what I should have done a long time ago -- I'll build traverse test/ and spawn workers for each test.
